qek/data/extractors.py

```diff
@@ -778,6 +828,134 @@ def run(self) -> CloudExtracted:
+        project_id: The ID of the project on the Pasqal Cloud API.
+        username: Your username on the Pasqal Cloud API.
+        password: Your password on the Pasqal Cloud API. If you leave
+            this to None, you will need to enter your password manually.
+        device_name: The name of the device to use. As of this writing,
+            the default value of "FRESNEL" represents the latest QPU
+            available through the Pasqal Cloud API.
+        batch_id: Use this to resume a workflow e.g. after turning off
+            your computer while the QPU was executing your sequences.
+    """
+
+    def __init__(
+        self,
+        compiler: BaseGraphCompiler[GraphType],
+        project_id: str,
+        username: str,
+        device_name: str = "FRESNEL",
+        password: str | None = None,
+        batch_ids: list[str] | None = None,
+        path: str | None = None,
+    ):
+        super().__init__(
+            compiler=compiler,
+            project_id=project_id,
+            username=username,
+            device_name=device_name,
+            password=password,
+            batch_ids=batch_ids,
+            path=path,
+        )
+
+    def run(self) -> PasqalCloudExtracted:
+        return self._run(emulator=None, config=None, state_extractor=lambda job, _seq: job.result)
+
+
+class RemoteEmuMPSExtractor(BaseRemoteExtractor[GraphType]):
+    """
+    An Extractor that uses the high-performance distributed emu-mps emulator
+    through Pasqal Cloud, to run sequences compiled from graphs.
+
+    Performance note (servers and interactive applications only):
+        If your code is meant to be executed as part of an interactive application or
+        a server, you should consider calling `await extracted` before your first call
+        to any of the methods of `extracted`. Otherwise, you will block the main thread.
+
+        If you are running this as part of an experiment, a Jupyter notebook, etc. you
+        may ignore this performance note.
+
+    Args:
+        path: Path to store the result of the run, for future uses.
+            To reload the result of a previous run, use `LoadExtractor`.
+        project_id: The ID of the project on the Pasqal Cloud API.
+        username: Your username on the Pasqal Cloud API.
+        password: Your password on the Pasqal Cloud API. If you leave
+            this to None, you will need to enter your password manually.
+        device_name: The name of the device to use. As of this writing,
+            the default value of "FRESNEL" represents the latest QPU
+            available through the Pasqal Cloud API.
+        batch_id: Use this to resume a workflow e.g. after turning off
+            your computer while the QPU was executing your sequences.
+    """
+
+    def __init__(
+        self,
+        compiler: BaseGraphCompiler[GraphType],
+        project_id: str,
+        username: str,
+        device_name: str = "FRESNEL",
```

Comment:
It doesn't seem the right ordering (wrt the extractor above). And should the `device_name` still be defaulted here ?

Reply:
Ordering fixed.

I believe that `device_name` should be defaulted, too. I expect that our users will not know the list of devices from the top of their head.